pytorch_lightning/utilities/distributed.py

```diff
@@ -302,6 +302,21 @@ def tpu_distributed() -> bool:
     return _TPU_AVAILABLE and xm.xrt_world_size() > 1
 
 
+def get_default_process_group_backend_for_device(device: torch.device) -> str:
+    return "nccl" if device.type == "cuda" else "gloo"
+
+
+def _get_process_group_backend_from_env() -> Optional[str]:
+    torch_backend = os.getenv("PL_TORCH_DISTRIBUTED_BACKEND")
+    if torch_backend:
+        rank_zero_deprecation(
+            "Environment variable `PL_TORCH_DISTRIBUTED_BACKEND`"
+            " was deprecated in v1.6 and will be removed in v1.8."
```

Comment:
What's the argument behind deprecating the environment variable?

Reply:
The motivation to specify the process group backend was from https://github.com/PyTorchLightning/pytorch-lightning/issues/5355 

The environment variable is not obvious or documented for users. https://github.com/PyTorchLightning/pytorch-lightning/pull/5981

Given it's now explicitly available on the constructor, why would someone use the env var over this argument?

Wouldn't this also be better supported from the LIghtning CLI, given there are other strategy arguments like `find_unused_parameters` that people pass through the constructor too? Why is one option set through a constructor argument and another through an env var?

I advocate deprecating this for consistency

Reply:
I would personally keep both with a preference toward the pg_backend argument, I believe it is quite useful for users to `pip install fairing` and add the flag and they don't need to modify their code.

Reply:
Yes, it does.

The `LightningCLI` supersedes all the argparse logic we had plus the saving part of `save_hyperparameters`, however, not everybody uses or wants to use the CLI so that's why none of that has been deprecated.